File: CsvSearch.cpp

#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main () {

    ifstream ip("data1.csv");

  if(!ip.is_open()) std::cout << "ERROR OPENING FILE" << '\n';

  string timestamp;
  string city;
  string temperature;
  string light;
  string airquality_raw;
  string sound;
  string humidity;
  string dust;
  string day;
  string month;
  string year;

   string cityinput;
   string dayinput;
   string monthinput;
   string yearinput;
    string choice;

    cout<<"Press 1 to search City\n";
    cout<<"Press 2 to search Day\n";
    cout<<"Press 3 to search Month\n";
    cout<<"Press 4 to search Year\n";

    cin>> choice;
   // cout <<choice;

    if (choice == "1"){
     cout<<"Enter City\n";
    cin>> cityinput;

  while(ip.good()){
    getline(ip,timestamp,',');
    getline(ip,city,',');
    getline(ip,temperature,',');
    getline(ip,light,',');
    getline(ip,airquality_raw,',');
    getline(ip,sound,',');
    getline(ip,humidity,',');
    getline(ip,dust,',');
    getline(ip,year, ',');
    getline(ip,month,',');
    getline(ip,day,'\n');

    if (cityinput == city)
        {
            std::cout<<timestamp <<" "<<city<< " " <<temperature<<" "<<light<<" " <<airquality_raw<<" "<<sound<<" "<<humidity<< " " << dust<< '\n';
        }
  }

  ip.close();
}


if (choice == "2"){
    cout<<"Enter Day\n";
    cin>> dayinput;

  while(ip.good()){
    getline(ip,timestamp,',');
    getline(ip,city,',');
    getline(ip,temperature,',');
    getline(ip,light,',');
    getline(ip,airquality_raw,',');
    getline(ip,sound,',');
    getline(ip,humidity,',');
    getline(ip,dust,',');
    getline(ip,year, ',');
    getline(ip,month,',');
    getline(ip,day,'\n');

    if (dayinput == day)
        {
            std::cout<<timestamp <<" "<<city<< " " <<temperature<<" "<<light<<" " <<airquality_raw<<" "<<sound<<" "<<humidity<< " " << dust<< '\n';
        }
  }

  ip.close();
}



if (choice == "3"){

     cout<<"Enter Month\n";
    cin>> monthinput;

  while(ip.good()){
    getline(ip,timestamp,',');
    getline(ip,city,',');
    getline(ip,temperature,',');
    getline(ip,light,',');
    getline(ip,airquality_raw,',');
    getline(ip,sound,',');
    getline(ip,humidity,',');
    getline(ip,dust,',');
    getline(ip,year, ',');
    getline(ip,month,',');
    getline(ip,day,'\n');

    if (monthinput == month)
        {
            std::cout<<timestamp <<" "<<city<< " " <<temperature<<" "<<light<<" " <<airquality_raw<<" "<<sound<<" "<<humidity<< " " << dust<< '\n';
        }
  }

  ip.close();
}


if (choice == "4"){
    cout<<"Enter Year\n";
    cin>> yearinput;

  while(ip.good()){
    getline(ip,timestamp,',');
    getline(ip,city,',');
    getline(ip,temperature,',');
    getline(ip,light,',');
    getline(ip,airquality_raw,',');
    getline(ip,sound,',');
    getline(ip,humidity,',');
    getline(ip,dust,',');
    getline(ip,year, ',');
    getline(ip,month,',');
    getline(ip,day,'\n');

    if (yearinput == year)
        {
            std::cout<<timestamp <<" "<<city<< " " <<temperature<<" "<<light<<" " <<airquality_raw<<" "<<sound<<" "<<humidity<< " " << dust<< '\n';
        }
  }

  ip.close();
}




}

File: CsvSearch.py
# -*- coding: utf-8 -*-
"""
Created on Mon Aug 27 01:54:48 2018

@author: Mossy
"""
import csv
def menu():
        
    print("Press 1 to Search by City")
    print("Press 2 to Search by Day")
    print("Press 3 to Search by Month")
    print("Press 4 to Search by Year")
    
    choice = input()
    if choice == "1":  
        cityinput = input("Enter City:")
        with open('data1.csv', 'rt') as f:
            reader = csv.reader(f, delimiter=',')
            for row in reader:
                if cityinput == row[1]: # if the username shall be on column 3 (-> index 2)
                    print(row)

    
    if choice == "2":  
        dayinput = input("Enter Day:")
        with open('data1.csv', 'rt') as f:
            reader = csv.reader(f, delimiter=',')
            for row in reader:
                if dayinput == row[10]: 
                    print(row)

    if choice == "3":  
        monthinput = input("Enter Month:")
        with open('data1.csv', 'rt') as f:
            reader = csv.reader(f, delimiter=',')
            for row in reader:
                if monthinput == row[9]: 
                    print(row)

    
    if choice == "4":  
        yearinput = input("Enter Year:")
        with open('data1.csv', 'rt') as f:
            reader = csv.reader(f, delimiter=',')
            for row in reader:
                if yearinput == row[8]: 
                    print(row)
menu()